plugins/modules/foreman_search_facts.py

```diff
@@ -84,18 +91,26 @@ def main():
         argument_spec=dict(
             resource=dict(type='str', required=True),
             search=dict(default=""),
+            full_details=dict(type='bool', aliases=['info'], default='false'),
         ),
     )
 
     module_params = module.clean_params()
-    entity = module_params['resource']
+    resource = module_params['resource']
     search = module_params['search']
 
     module.connect()
 
-    response = module.list_resource(entity, search)
+    response = module.list_resource(resource, search)
+
+    if module_params['full_details']:
+        resources = []
+        for res in response:
+            resources.append(module.show_resource(resource, res['id']))
```

Comment:
Why not as a comprehension (modern python)?
`resources = [module.show_resource(resource, res['id'])) for res in response]`
Nice how it it unclear whether res is the abbreviation of resource or response. :smiling_imp:

Reply:
> Nice how it it unclear whether res is the abbreviation of resource or response.

Damn, you noticed it :stuck_out_tongue_closed_eyes: